cmd/weaviate/main.go

```diff
@@ -0,0 +1,328 @@
+//                           _       _
+// __      _____  __ ___   ___  __ _| |_ ___
+// \ \ /\ / / _ \/ _` \ \ / / |/ _` | __/ _ \
+//  \ V  V /  __/ (_| |\ V /| | (_| | ||  __/
+//   \_/\_/ \___|\__,_| \_/ |_|\__,_|\__\___|
+//
+//  Copyright © 2016 - 2024 Weaviate B.V. All rights reserved.
+//
+//  CONTACT: hello@weaviate.io
+//
+
+package main
+
+import (
+	"fmt"
+	"net"
+	"net/http"
+	"path/filepath"
+	"runtime/debug"
+	"strings"
+	"time"
+
+	"github.com/jessevdk/go-flags"
+	"github.com/sirupsen/logrus"
+	"github.com/weaviate/weaviate/adapters/clients"
+	"github.com/weaviate/weaviate/adapters/repos/db"
+	"github.com/weaviate/weaviate/adapters/repos/db/inverted"
+	clusterweaviate "github.com/weaviate/weaviate/cluster"
+	"github.com/weaviate/weaviate/entities/replication"
+	"github.com/weaviate/weaviate/entities/vectorindex"
+	"github.com/weaviate/weaviate/exp/query"
+	"github.com/weaviate/weaviate/grpc/generated/protocol/v1"
+	"github.com/weaviate/weaviate/usecases/auth/authentication/apikey"
+	"github.com/weaviate/weaviate/usecases/auth/authentication/composer"
+	"github.com/weaviate/weaviate/usecases/auth/authentication/oidc"
+	"github.com/weaviate/weaviate/usecases/auth/authorization"
+	"github.com/weaviate/weaviate/usecases/cluster"
+	"github.com/weaviate/weaviate/usecases/config"
+	"github.com/weaviate/weaviate/usecases/memwatch"
+	"github.com/weaviate/weaviate/usecases/modules"
+	"github.com/weaviate/weaviate/usecases/objects"
+	"github.com/weaviate/weaviate/usecases/scaler"
+	"github.com/weaviate/weaviate/usecases/schema"
+	"github.com/weaviate/weaviate/usecases/traverser"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/reflection"
+
+	schemarepo "github.com/weaviate/weaviate/adapters/repos/schema"
+)
+
+const (
+	TargetQuerier = "querier"
+
+	// For mocking purpose
+	authAnonymousEnabled = true
+)
+
+// TODO: We want this to be part of original `cmd/weaviate-server`.
+// But for some reason that binary is auto-generated and I couldn't modify as I need. Hence separate binary for now
+func main() {
+	var opts Options
+	log := logrus.WithFields(logrus.Fields{"app": "weaviate"}).Logger
```

Comment:
nit: we can adjust that "app=XX" depending on if it's querier/importer/etc... later down the line.

Reply:
actually we are adding `target=querier` here.
https://github.com/weaviate/weaviate/blob/fb6017fb3763edcaecefbe1b75932dfde2bafd3b/cmd/weaviate/main.go#L234

It would look like `app=weaviate, target=querier` and `target=xxxx` changes depends on the component.